lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -1335,12 +1335,14 @@ namespace Js
             EngineInterfaceObject* nativeEngineInterfaceObj = scriptContext->GetLibrary()->GetEngineInterfaceObject();
             if (nativeEngineInterfaceObj)
             {
-                IntlEngineInterfaceExtensionObject* intlExtensionObject = static_cast<IntlEngineInterfaceExtensionObject*>(nativeEngineInterfaceObj->GetEngineExtension(EngineInterfaceExtensionKind_Intl));
+                IntlEngineInterfaceExtensionObject* intlExtensionObject = static_cast<IntlEngineInterfaceExtensionObject*>(
+                    nativeEngineInterfaceObj->GetEngineExtension(EngineInterfaceExtensionKind_Intl));
                 if (args.Info.Count == 2)
                 {
                     auto undefined = scriptContext->GetLibrary()->GetUndefined();
                     CallInfo toPass(callInfo.Flags, 7);
-                    return intlExtensionObject->EntryIntl_CompareString(function, toPass, undefined, pThis, pThat, undefined, undefined, undefined, undefined);
+                    return CALL_ENTRYPOINT(scriptContext->GetThreadContext(), intlExtensionObject->EntryIntl_CompareString, function, toPass,
+                        undefined, pThis, pThat, undefined, undefined, undefined, undefined);
```

Comment:
nit: have the params to `intlExtensionObject->EntryIntl_CompareString` all together on the second line.

Reply:
@rajatd Does CALL_ENTRYPOINT need to be wrapped in implicit call like the uses in JavascriptOperators.cpp (like CallGetter)

Pattern:

```
...
return threadContext->ExecuteImplicitCall(function, ImplicitCall_Accessor, [=]() -> Js::Var
{
...
Var result = CALL_ENTRYPOINT(...)
result = CrossSite::MarshalVar(requestContext, result);
return result;
});
```

Reply:
@jackhorton I think some carefully-designed tests with a loop and `-mic:1` might reveal JIT-related failures.

Reply:
@dilijev I am not sure what careful design would need to go in here to execute anything like that. Also, keep in mind that Intl is not enabled in xplat CI (or else we would have seen this, most likely, from the single test we have that calls .localeCompare), so tests stressing it wont reveal much for now (this bug is xplat-only because CALL_ENTRYPOINT is only "needed" on xplat, modulo your concern about implicit calls.